### PR TITLE
Added gui and cscope features to Vim package

### PR DESCRIPTION
**Pull request by alfredo-gimenez**

### CODE REVIEW

**Comment by trws** [comment]
Nice!

One nit though, if you're putting in a variant for cscope, don't we need a cscope package for it to depend on?

Once my PR for flux dependencies, #92,  lands lua support can go in as well.

**Comment by alfredo-gimenez** [comment]
Oh, right! Currently it just finds the installed cscope. I'll add a cscope package to the PR.

**Comment by alfredo-gimenez** [comment]
Added

**Comment by trws** [comment]
:+1: Once this gets merged in I'll add cscope support to the global tags package too.

File: var/spack/packages/vim/package.py [hunk 1]
```diff
@@ -44,6 +50,11 @@ def install(self, spec, prefix):
               feature_set,
               fs))
           feature_set = fs
+      if '+gui' in spec:
+        if feature_set is not None:
+          if feature_set is not 'huge':
+            tty.error("+gui variant requires 'huge' feature set, {} was specified".format(feature_set))
```
Comment (tgamblin) [inline var/spack/packages/vim/package.py]: @alfredo-gimenez: You can't use {} in a format expression in 2.6 -- this should break one of the tests so be sure to run `spack test` before committing and it'll catch this.  I'll merge and fix this time.

**Pull request merged**